src/Table/Table.jsx

```diff
@@ -0,0 +1,132 @@
+import lodash from "lodash";
+import React, {Component, PropTypes} from "react";
+
+import * as tablePropTypes from "./tablePropTypes";
+import Cell from "./Cell";
+import Column from "./Column";
+import Header from "./Header";
+import sortDirection from "./sortDirection";
+import {joinClasses} from "../utils";
+
+require("./Table.less");
+
+
+export class Table extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sortedData: props.data,
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.initialSortState) {
+      this._sort(this.props.initialSortState);
+    }
+  }
+
+  _toggleSort(columnIndex) {
+    const {sortedData, sortState} = this.state;
+
+    if (sortState && sortState.columnIndex === columnIndex) {
+      sortState.direction =
+        sortState.direction === sortDirection.ASCENDING ? sortDirection.DESCENDING : sortDirection.ASCENDING;
+      this.setState({sortState, sortedData: sortedData.reverse()});
+      this.props.onSortChange(sortState);
+      return;
+    }
+
+    this._sort({
+      columnIndex,
+      direction: sortDirection.ASCENDING,
+    });
+  }
+
+  _sort(sortState) {
+    const columns = this.props.children;
+
+    if (typeof sortState.columnIndex !== "number" && sortState.columnID) {
+      sortState.columnIndex = lodash.findIndex(columns, column => column.props.id === sortState.columnID);
+    }
+
+    const sortedColumn = this.props.children[sortState.columnIndex];
+    if (!sortedColumn || !sortedColumn.props.sortable) {
+      return;
+    }
+
+    let sortedData = lodash(this.props.data)
+      .sortBy(row => {
+        let value = sortedColumn.props.sortValueFn(row);
+
+        if (typeof value === "string") {
+          value = value.trim().toLowerCase();
+        }
+
+        return value;
+      });
+
+    if (sortState.direction === sortDirection.DESCENDING) {
+      sortedData = sortedData.reverse();
+    }
+
+    sortState.columnID = sortedColumn.props.id;
+    this.setState({sortState, sortedData: sortedData.value()});
+    this.props.onSortChange(sortState);
+  }
+
+  render() {
+    const {children, className, filter, fixed, rowIDFn} = this.props;
+    const {sortedData, sortState} = this.state;
+    const {cssClass} = Table;
+
+    return (
+      <table className={joinClasses(cssClass.TABLE, fixed && cssClass.FIXED, className)}>
+        <Header onSortChange={columnIndex => this._toggleSort(columnIndex)} sortState={sortState}>
+          {children}
+        </Header>
+        <tbody className={cssClass.BODY}>
+          {sortedData.map((rowData, rowIndex) => (filter(rowData) ? (
+            <tr className={cssClass.ROW} key={rowIDFn ? rowIDFn(rowData) : rowIndex}>
```

Comment:
Is that really true though? I know there's really weird stuff that happens when the keys don't actually uniquely identify content (like re-renders not occurring properly). I've had to deal with some devilish bugs because of keys that weren't correctly unique.